### PR TITLE
command package changes, service unit tests, example program modified

### PR DESCRIPTION
**Pull request by milosgajdos83**
This PR introduces some basic service package tests. They still need
some refactoring, but this is a good start.

Some data structures (`command.Commander` and `command.Outer`
interfaces) in `command` package have been renamed to make them more
digestible and imo more semantically correct and consistent with other
similar data structures in the project. The same has been done for
`service.OsService`.

Example program now shows how to build a simple `go` binary using
`servpeek` packages. Getting rid of `_test.go` files in examples dir
also allows for running `go test ./...` in the project root which is the
first step to integration with Circle/Travis.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ Project currently provides support for various package managers: `apt`, `yum`, `
 
 # Usage
 
-Currently there is no single program to run as it stands. Instead you can simply invoke the familiar `go test` on any of the examples in `examples` subdirectory. You can obviously create your own `_test.go` files for the stuff you want to test.
+Currently there is no single program to run as it stands. Instead the project provides API to build your own go binaries that can be used to test your infrastrucure - this way there is no need to have any runtime available on the tested infrastructure, just your binary. You can obviously create your own `_test.go` files for the stuff you want to test, but you will need to have go tool chain available on the tested server. Building binaries is recommended and preferrable option.
```
Comment (agonzalezro) [inline README.md]: What do you mean with: "but you will need to have go tool chain available on the tested server. Building binaries is recommended and preferrable option". This is not really true.

File: README.md [hunk 2]
```diff
@@ -27,35 +27,9 @@ Get the package:
 $ go get github.com/milosgajdos83/servpeek
 ```
 
-Example test file could look like this:
+You can find an example program in the [examples](https://github.com/milosgajdos83/servpeek/tree/master/examples) directory.
 
-```go
-package example
+## TODO
```
Comment (agonzalezro) [inline README.md]: I wouldn't remove all the examples. I would show one or two here and then point the people to the `examples/` folder. BTW, I think that just pointing like: `[examples](examples/)` will work, but... don't really trust me on this one.

File: pkg/mocks_test.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
+package pkg
```
Comment (agonzalezro) [inline pkg/mocks_test.go]: I wouldn't call it `_test.go` since it's not testing anything.

File: service/matchers_test.go [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+	statuses := []string{"running", "stopped"}
+	for _, status := range statuses {
+		for _, svc := range svcs {
+			currentDir, err := os.Getwd()
+			assert.NoError(err)
+			fixturesPath := path.Join(currentDir, "test-fixtures",
+				svc.sysInitType+"-"+status+".out")
+			cmdOut, err := ioutil.ReadFile(fixturesPath)
+			assert.NoError(err)
+			s := &mockSvc{
```
Comment (agonzalezro) [inline service/matchers_test.go]: I would move this to: `newMockSvc` to clarify this test a little bit.

File: service/matchers_test.go [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+		{"docker", "upstart"},
+		{"docker", "systemd"},
+	}
+
+	statuses := []string{"running", "stopped"}
+	for _, status := range statuses {
+		for _, svc := range svcs {
+			currentDir, err := os.Getwd()
+			assert.NoError(err)
+			fixturesPath := path.Join(currentDir, "test-fixtures",
```
Comment (agonzalezro) [inline service/matchers_test.go]:
```
file := fmt.Sprintf("%s-%s.out", svc.sysInitType, status)
fixturesPath := path.Join(currentDir, "test-fixtures", file)
```

Or:

```
fixturesPath := getFixturePath(svc.sysInitType, status)
```

With the latest you will get rid of `currentDir :=...` and the other `assert`.

File: service/matchers_test.go [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+				svc.sysInitType+"-"+status+".out")
+			cmdOut, err := ioutil.ReadFile(fixturesPath)
```
Comment (agonzalezro) [inline service/matchers_test.go]: Forget about the previous message: `fixture := getFixture(svc.sysInitType, status)`.

You will get rid of all this setup to get the fixture.

File: service/service.go [hunk 7]
```diff
@@ -38,43 +34,42 @@ func (s Status) String() string {
...
+// NewSvc creates new Service or returns error if the requested service type is not supported
+func NewSvc(name, sysInitType string) (*Svc, error) {
+	sysInit, err := NewSysInit(sysInitType)
+	if err != nil {
+		return nil, err
 	}
 
-	if !supported[sysInitType] {
-		return nil, fmt.Errorf("Unsupported service init type: %s", sysInitType)
+	if name == "" {
```
Comment (agonzalezro) [inline service/service.go]: I wouldn't test this. The "constructor" is expecting a string, if you send an empty string you know what you are doing, or... you want to break something.

File: README.md (hunk 1 above)
Comment (milosgajdos83) [inline README.md]: well if you ship `_test.go` files you need go tool chain to be able to execute `go test`. Am I wrong ? The "preferable option" bit - this needs rewording I agree, but if you ship your "testing" binary you don't need go to be installed on the tested server.

File: pkg/mocks_test.go (hunk 3 above)
Comment (milosgajdos83) [inline pkg/mocks_test.go]: But it belongs to the tests - I could easily move any of the data structures defined here to any of the `_test.go` files where they are used. That was the sole reason why I've renamed this file as functionally, it does not serve any purpose other than for testing.

File: service/matchers_test.go (hunk 5 above)
Comment (milosgajdos83) [inline service/matchers_test.go]: Excellent point! :+1:

File: service/matchers_test.go (hunk 6 above)
Comment (milosgajdos83) [inline service/matchers_test.go]: Actually, this IS better :-)

File: service/service.go (hunk 7 above)
Comment (milosgajdos83) [inline service/service.go]: Good point. Thanks.

File: README.md (hunk 1 above)
Comment (agonzalezro) [inline README.md]: Okkkkk I got it now! Yes please, rephrase it :D

File: pkg/mocks_test.go (hunk 3 above)
Comment (agonzalezro) [inline pkg/mocks_test.go]: But `go test` will pick this file as testable and it will not find any test.

Comment (milosgajdos83) [inline pkg/mocks_test.go]: True, but it seems `go test` does not complain about it so :-) But yeah I agree with you on this - I just don't know where does it belong :-/

File: service/matchers_test.go (hunk 4 above)
Comment (milosgajdos83) [inline service/matchers_test.go]: Yes, excellent idea! Maybe I should do the same for `mockPkg`.

File: service/service.go (hunk 7 above)
Comment (milosgajdos83) [inline service/service.go]: Let's keep this one in actually and do a bit of "defensive" programming :-) There isn't much wrong here keeping it in imo.

**Comment by agonzalezro** [comment]
Remember to add CircleCI and the badge to the README :)

**Pull request merged**